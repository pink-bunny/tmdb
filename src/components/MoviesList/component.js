import React from 'react';
import PropTypes from 'prop-types';
import {
  Row,
  Col,
  Pagination,
  Empty,
  Typography,
  Spin
} from 'antd';

import Movie from '../MovieItem';

const MoviesList = ({
  list,
  loading,
  error,
  totalItems,
  onPaginationClick,
  currentPage,
  emptyListTxt,
  actionsList
}) => {
  if (loading) {
    return (
      <Row type="flex" justify="center">
        <Col>
          <Spin />
        </Col>
      </Row>
    );
  }

  if (error) {
    return (
      <Row type="flex" justify="center">
        <Col>
          <Typography.Paragraph type="danger">
            {error}
          </Typography.Paragraph>
        </Col>
      </Row>
    );
  }

  if (!list) {
    return (
      <Empty
        description={emptyListTxt}
        image={Empty.PRESENTED_IMAGE_SIMPLE}
      />
    );
  }

  return (
    <>
      <Row
        type="flex"
        gutter={16}
      >
        <Col
          span={20}
          offset={2}
        >
          {list.map((item) => (
            <Col
              key={item.id}
              xs={{ span: 12 }}
              sm={{ span: 12 }}
              md={{ span: 8 }}
              lg={{ span: 6 }}
              xl={{ span: 4 }}
            >
              <Movie
                id={item.id}
                title={item.title}
                overview={item.overview}
                poster={item.poster_path}
                actions={actionsList(item.id, item.title)}
              />
            </Col>
          ))}
        </Col>
      </Row>
      {totalItems > 20 ? (
        <Row
          type="flex"
          justify="center"
        >
          <Col>
            <Pagination
              current={currentPage}
              pageSize={20}
              total={totalItems}
              className="pagination"
              onChange={onPaginationClick}
            />
          </Col>
        </Row>
      ) : (
        null
      )}
    </>
  );
};

MoviesList.propTypes = {
  actionsList: PropTypes.func,
  list: PropTypes.arrayOf(PropTypes.object),
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  totalItems: PropTypes.number.isRequired,
  onPaginationClick: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
  emptyListTxt: PropTypes.string.isRequired
};

MoviesList.defaultProps = {
  actionsList: () => [],
  list: null,
  error: null
};

export default MoviesList;
