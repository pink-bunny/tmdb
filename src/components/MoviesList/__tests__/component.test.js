import React from 'react';
import { shallow } from 'enzyme';

import MoviesList from '../component';

describe('MoviesList component', () => {
  const requiredProps = {
    loading: false,
    error: '',
    totalItems: 0,
    onPaginationClick: jest.fn(),
    currentPage: 0,
    emptyListTxt: 'No movies found'
  };
  const listArr = [{
    id: 1,
    title: 'Terminator: Dark Fate',
    overview: 'Decades after Sarah Connor prevented Judgment Day.',
    poster_path: '/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg' /* eslint-disable-line */
  }];

  it('matches snapshot when it is loading', () => {
    const props = {
      ...requiredProps,
      loading: true
    };
    const wrapper = shallow(<MoviesList {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when it has error', () => {
    const props = {
      ...requiredProps,
      error: 'Loading error'
    };
    const wrapper = shallow(<MoviesList {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with empty list', () => {
    const props = {
      ...requiredProps,
      list: null
    };
    const wrapper = shallow(<MoviesList {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with list and without pagination', () => {
    const props = {
      ...requiredProps,
      list: listArr
    };
    const wrapper = shallow(<MoviesList {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with list and pagination', () => {
    const props = {
      ...requiredProps,
      list: listArr,
      currentPage: 1,
      totalItems: 21
    };
    const wrapper = shallow(<MoviesList {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
