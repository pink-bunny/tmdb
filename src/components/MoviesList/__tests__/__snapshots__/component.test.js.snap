// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component matches snapshot when it has error 1`] = `
<Row
  gutter={0}
  justify="center"
  type="flex"
>
  <Col>
    <Paragraph
      type="danger"
    >
      Loading error
    </Paragraph>
  </Col>
</Row>
`;

exports[`MoviesList component matches snapshot when it is loading 1`] = `
<Row
  gutter={0}
  justify="center"
  type="flex"
>
  <Col>
    <Spin
      size="default"
      spinning={true}
      wrapperClassName=""
    />
  </Col>
</Row>
`;

exports[`MoviesList component matches snapshot with empty list 1`] = `
<Empty
  description="No movies found"
  image={<Simple />}
/>
`;

exports[`MoviesList component matches snapshot with list and pagination 1`] = `
<Fragment>
  <Row
    gutter={16}
    type="flex"
  >
    <Col
      offset={2}
      span={20}
    >
      <Col
        key="1"
        lg={
          Object {
            "span": 6,
          }
        }
        md={
          Object {
            "span": 8,
          }
        }
        sm={
          Object {
            "span": 12,
          }
        }
        xl={
          Object {
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <Movie
          actions={Array []}
          id={1}
          overview="Decades after Sarah Connor prevented Judgment Day."
          poster="/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg"
          title="Terminator: Dark Fate"
        />
      </Col>
    </Col>
  </Row>
  <Row
    gutter={0}
    justify="center"
    type="flex"
  >
    <Col>
      <Pagination
        className="pagination"
        current={1}
        onChange={[MockFunction]}
        pageSize={20}
        total={21}
      />
    </Col>
  </Row>
</Fragment>
`;

exports[`MoviesList component matches snapshot with list and without pagination 1`] = `
<Fragment>
  <Row
    gutter={16}
    type="flex"
  >
    <Col
      offset={2}
      span={20}
    >
      <Col
        key="1"
        lg={
          Object {
            "span": 6,
          }
        }
        md={
          Object {
            "span": 8,
          }
        }
        sm={
          Object {
            "span": 12,
          }
        }
        xl={
          Object {
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <Movie
          actions={Array []}
          id={1}
          overview="Decades after Sarah Connor prevented Judgment Day."
          poster="/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg"
          title="Terminator: Dark Fate"
        />
      </Col>
    </Col>
  </Row>
</Fragment>
`;
